lightning/src/chain/onchaintx.rs

```diff
@@ -292,24 +296,7 @@ impl<'a, K: KeysInterface> ReadableArgs<&'a K> for OnchainTxHandler<K::Signer> {
 		let waiting_threshold_conf_len: u64 = Readable::read(reader)?;
 		let mut onchain_events_awaiting_threshold_conf = Vec::with_capacity(cmp::min(waiting_threshold_conf_len as usize, MAX_ALLOC_SIZE / 128));
 		for _ in 0..waiting_threshold_conf_len {
-			let txid = Readable::read(reader)?;
-			let height = Readable::read(reader)?;
-			let event = match <u8 as Readable>::read(reader)? {
-				0 => {
-					let claim_request = Readable::read(reader)?;
-					OnchainEvent::Claim {
-						claim_request
-					}
-				},
-				1 => {
-					let package = Readable::read(reader)?;
-					OnchainEvent::ContentiousOutpoint {
-						package
-					}
-				}
-				_ => return Err(DecodeError::InvalidValue),
-			};
-			onchain_events_awaiting_threshold_conf.push(OnchainEventEntry { txid, height, event });
+			onchain_events_awaiting_threshold_conf.push(Readable::read(reader)?);
 		}
 
 		read_tlv_fields!(reader, {}, {});
```

Comment:
Should `read_tlv_fields`/`write_tlv_fields` also have a separate `vecfield` `{}`-group at the end?

Reply:
Hmmm, I suppose we could rejigger them a bit, currently they rely on the user having variables for each field already defined, which would be a big change to lean on the `VecReadWrapper`. Lets plan on cleaning a few things up in a followup?